include/boost/geometry/algorithms/detail/is_simple/linear.hpp

```diff
@@ -54,128 +57,210 @@ namespace detail { namespace is_simple
 {
 
 
-template <typename Linestring, bool CheckSelfIntersections = true>
-struct is_simple_linestring
+template <typename Turn>
+inline bool check_segment_indices(Turn const& turn,
+                                  signed_index_type last_index)
 {
-    static inline bool apply(Linestring const& linestring)
+    return
+        (turn.operations[0].seg_id.segment_index == 0
+         && turn.operations[1].seg_id.segment_index == last_index)
+        ||
+        (turn.operations[0].seg_id.segment_index == 0
+         && turn.operations[1].seg_id.segment_index == last_index);
+}
+
+
+template <typename Geometry, typename Tag = typename tag<Geometry>::type>
+class is_acceptable_turn
+    : not_implemented<Geometry>
+{};
+
+template <typename Linestring>
+class is_acceptable_turn<Linestring, linestring_tag>
+{
+public:
+    is_acceptable_turn(Linestring const& linestring)
+        : m_linestring(linestring)
+        , m_is_closed(geometry::equals(range::front(linestring),
+                                       range::back(linestring)))
+    {}
+
+    template <typename Turn>
+    inline bool apply(Turn const& turn) const
     {
-        return !detail::is_valid::has_duplicates
-                    <
-                        Linestring, closed
-                    >::apply(linestring)
-            && !detail::is_valid::has_spikes
-                    <
-                        Linestring, closed
-                    >::apply(linestring)
-            && !(CheckSelfIntersections && geometry::intersects(linestring));
+        BOOST_ASSERT(boost::size(m_linestring) > 1);
+        return m_is_closed
+            && turn.method == overlay::method_none
+            && check_segment_indices(turn, boost::size(m_linestring) - 2)
+            && geometry::equals(range::front(m_linestring), turn.point);
```

Comment:
Same answer: I will try to replace the call to `geometry::equals` by means of the segment ratios.